bin/dev-server.js

```diff
@@ -82,23 +82,6 @@ function watchAll() {
     debounce(rebuildPerf, 700, {leading: true}));
 }
 
-var filesWritten = false;
-
-Promise.resolve().then(function () {
-  if (process.env.TRAVIS) {
-    return; // don't bother rebuilding in Travis; we already built
-  }
-  return Promise.all([
-    rebuildPouch(),
-    rebuildTestUtils(),
-    rebuildPerf()
-  ]);
-}).then(function () {
-  console.log('Rebuilt PouchDB/test/perf JS bundles');
-  filesWritten = true;
-  checkReady();
-});
```

Comment:
Does removing this break hot-reloading when running the dev server locally?

Reply:
I've moved the initial build to bin/run-dev.sh.

To better understand this change, here's the call order, when `npm run dev` is executed:
1. npm run dev -> calls ./bin/run-dev.sh
2. run-dev.sh -> builds via build-pouchdb.js and npm build-test (equivalent to the removed build promise you mentioned) and executes run-test.sh
3. run-test.sh -> invokes npm run launch-dev-server
4. npm run launch-dev-server finally run `node ./bin/dev-server.js`

(I know exactly what you think: it's a mess)

My thought: `bin/dev-server.js` is an implementation detail and isn't meant to invoke directly (and not mentioned in TESTING.md). Also: build artifacts aren't required to start `dev-server.js`.

Reply:
OK, I understand re: hot-reloading.

Reply:
> Edit: Now I'm thinking; should we separate "utilities" into a sub-folder of bin?
> This makes it clearer which files are intended for execution by users.

If this is planned, I'd prefer it as a separate PR - there's already a lot going on here

Reply:
> If this is planned, I'd prefer it as a separate PR - there's already a lot going on here

Agree 👍️